x-pack/plugins/lens/public/editor_frame_service/error_helper.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { ExpressionRenderError } from 'src/plugins/expressions/public';
+
+interface ElasticsearchErrorClause {
+  type: string;
+  reason: string;
+  caused_by?: ElasticsearchErrorClause;
+}
+
+interface RequestError extends Error {
+  body?: { attributes?: { error: ElasticsearchErrorClause } };
+}
+
+const isRequestError = (e: Error | RequestError): e is RequestError => {
+  if ('body' in e) {
+    return e.body?.attributes?.error?.caused_by !== undefined;
+  }
+  return false;
+};
+
+function getNestedErrorClause({
+  type,
+  reason,
+  caused_by: causedBy,
+}: ElasticsearchErrorClause): { type: string; reason: string } {
+  if (causedBy) {
+    return getNestedErrorClause(causedBy);
+  }
+  return { type, reason };
```

Comment:
Should we be skipping all the intermediate types and reasons? This recursion makes sense, but are there cases where it's losing important information?

Reply:
I see the point - for the known cases the root exception is the relevant one and I made this experience very often in Java in general (lot's of re-thrown exceptions, but the actual cause is in the very last one). I guess when we solve this cleanly there should be an option to expand the error to show all the details. I will open an issue for this.